tapchannel/aux_invoice_manager_test.go

```diff
@@ -0,0 +1,571 @@
+			},
+		},
+	}
+
+	for _, testCase := range testCases {
+		testCase := testCase
+
+		t.Logf("Running AuxInvoiceManager test case: %v", testCase.name)
+
+		// Instantiate mock rfq manager.
+		mockRfq := &mockRfqManager{
+			peerBuyQuotes:   testCase.buyQuotes,
+			localSellQuotes: testCase.sellQuotes,
+		}
+
+		done := make(chan bool)
+
+		// Instantiate mock htlc modifier.
+		mockModifier := &mockHtlcModifier{
+			requestQue:     testCase.requests,
+			expectedResQue: testCase.responses,
+			done:           done,
+			t:              t,
+		}
+
+		// Create the manager.
+		manager := NewAuxInvoiceManager(
+			&InvoiceManagerConfig{
+				ChainParams:         testChainParams,
+				InvoiceHtlcModifier: mockModifier,
+				RfqManager:          mockRfq,
+			},
+		)
+
+		err := manager.Start()
+		require.NoError(t, err)
+
+		// If the manager is not done processing the htlc modification
+		// requests within the specified timeout, assume this is a
+		// failure.
+		select {
+		case <-done:
+		case <-time.After(testTimeout):
+			t.Fail()
+		}
+	}
+}
+
+// genRandomRfqID generates a random rfqmsg.ID value.
+func genRandomRfqID(t *rapid.T) rfqmsg.ID {
+	var rfqID rfqmsg.ID
+	for i := 0; i < len(rfqID); i++ {
+		rfqID[i] = rapid.Byte().Draw(t, "scid_byte")
+		// rapid.SliceOfN(rapid.Byte(), len(rfqID), len(rfqID))
+	}
+	return rfqID
+}
+
+// genInvoice generates an invoice that may have a random amount, and may have
+// routing hints.
+func genInvoice(t *rapid.T) *lnrpc.Invoice {
+	res := &lnrpc.Invoice{}
+
+	// Do a random draw with a 1/8 chance of introducing an empty invoice.
+	nullInv := rapid.Uint16().Draw(t, "invoice_exists")
+	if nullInv%8 == 0 {
+		return res
+	}
+
+	// Generate a random invoice value.
+	valueMsat := rapid.Int64Range(
+		0, 100_000_000,
+	).Draw(t, "invoice_value_msat")
+
+	res.ValueMsat = valueMsat
+
+	// Do a random draw with a 1/8 chance to introduce empty routing hints.
+	nullHints := rapid.Uint16().Draw(t, "invoice_hints_exist")
+	if nullHints%8 == 0 {
+		return res
+	}
+
+	res.RouteHints = testRouteHints()
+
+	return res
+}
+
+// genCustomRecords generates custom records that have a random amount of random
+// asset units, and may have an scid as routing hint.
+func genCustomRecords(t *rapid.T,
+	rfqID rfqmsg.ID) (lnwire.CustomRecords, uint64) {
+
+	assetUnits := rapid.Uint64Range(10_000, 100_000).Draw(t, "asset_units")
+
+	balance := []*rfqmsg.AssetBalance{
+		rfqmsg.NewAssetBalance(
+			assetID(rapid.Byte().Draw(t, "asset_id")), assetUnits,
+		),
+	}
+
+	htlc := genHtlc(t, balance, rfqID)
+
+	customRecords, err := lnwire.ParseCustomRecords(htlc.Bytes())
+	require.NoError(t, err)
+
+	return customRecords, assetUnits
+}
+
+// genHtlc generates an istance of rfqmsg.Htlc with the provided asset amounts
+// and rfqID.
+func genHtlc(t *rapid.T, balance []*rfqmsg.AssetBalance,
+	rfqID rfqmsg.ID) *rfqmsg.Htlc {
+
+	// Introduce a 1/5 chance of no rfqID in this htlc.
+	noRfqID := rapid.Uint16().Draw(t, "has_rfqid_probability")
```

Comment:
Same notes re: replacing `Uint16()` and mod with `Bool()`.

Reply:
https://github.com/lightninglabs/taproot-assets/pull/1144#discussion_r1813925080